File: Lab/SEM4/MPP/lab4/moto_java/src/main/java/org/zdroba/entity/Racer.java
package org.zdroba.entity;

public class Racer {
    private Long id;
    private String name;
    private String cnp;
    private RaceEvent engine;
    private Team team;

    public Racer(String name, String cnp, RaceEvent engine) {
        this.name = name;
        this.cnp = cnp;
        this.engine = engine;
        this.team = Team.NONE;
    }

    public Racer(String name, String cnp, Team team, RaceEvent engine) {
        this.name = name;
        this.cnp = cnp;
        this.team = team;
        this.engine = engine;
    }

    public Racer() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCnp() {
        return cnp;
    }

    public void setCnp(String cnp) {
        this.cnp = cnp;
    }

    public RaceEvent getEngine() {
        return engine;
    }

    public void setEngine(RaceEvent engine) {
        this.engine = engine;
    }

    public Team getTeam() {
        return team;
    }

    public void setTeam(Team team) {
        this.team = team;
    }
}

File: Lab/SEM4/MPP/lab4/moto_c/entity/Racer.cs
namespace moto_c.entity;

public class Racer
{
    public long id { get; set; }
    public string name { get; set; }
    public string cnp  { get; set; }
    public Team team { get; set; }
    public RaceEvent engine { get; set; }

    public Racer(string name, string cnp, RaceEvent engine)
    {
        this.name = name;
        this.cnp = cnp;
        this.engine = engine;
        this.team = Team.NONE;
    }
    
    public Racer()
    {}

    public Racer(string name, string cnp, Team team, RaceEvent engine)
    {
        this.name = name;
        this.cnp = cnp;
        this.team = team;
        this.engine = engine;
    }
}